File: web/src/main/java/no/nav/modiapersonoversikt/legacy/sak/config/SakServiceConfig.java
package no.nav.modiapersonoversikt.legacy.sak.config;

import no.nav.modiapersonoversikt.config.service.EnhetligKodeverkConfig;
import no.nav.modiapersonoversikt.legacy.sak.service.*;
import no.nav.modiapersonoversikt.legacy.sak.service.filter.Filter;
import no.nav.modiapersonoversikt.legacy.sak.service.saf.ExperimentSafService;
import no.nav.modiapersonoversikt.legacy.sak.service.saf.SafGraphqlServiceImpl;
import no.nav.modiapersonoversikt.legacy.sak.service.saf.SafService;
import no.nav.modiapersonoversikt.legacy.sak.service.saf.SafServiceImpl;
import no.nav.modiapersonoversikt.legacy.sak.utils.TemagrupperHenter;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

@Configuration
@Import({EnhetligKodeverkConfig.class})
public class SakServiceConfig {

    @Bean
    public Filter sakOgBehandlingFilter() {
        return new Filter();
    }

    @Bean
    public SakOgBehandlingService sakOgBehandlingService() {
        return new SakOgBehandlingService();
    }

    @Bean
    public PesysService pesysService() {
        return new PesysService();
    }

    @Bean
    public SakstemaService sakstemaService() {
        return new SakstemaService();
    }

    @Bean
    public SaksService saksService() {
        return new SaksService();
    }

    @Bean
    public DokumentMetadataService dokumentMetadataService(SafService safService) {
        return new DokumentMetadataService(safService);
    }

    @Bean
    public TemagrupperHenter temagrupperHenter() {
        return new TemagrupperHenter();
    }

    @Bean
    public SakstemaGrupperer sakstemaGrupperer() {
        return new SakstemaGrupperer();
    }

    @Bean
    public BulletproofKodeverkService bulletproofKodeverkService() {
        return new BulletproofKodeverkService();
    }

    @Bean
    public SafService safService() {
        return new ExperimentSafService(
                new SafServiceImpl(),
                new SafGraphqlServiceImpl()
        );
    }
}

File: web/src/main/java/no/nav/modiapersonoversikt/legacy/sak/SakServiceConfig.kt
package no.nav.modiapersonoversikt.legacy.sak

import no.nav.modiapersonoversikt.consumer.sakogbehandling.SakOgBehandlingService
import no.nav.modiapersonoversikt.consumer.sakogbehandling.SakOgBehandlingServiceImpl
import no.nav.modiapersonoversikt.legacy.sak.service.SakstemaService
import no.nav.modiapersonoversikt.legacy.sak.service.TilgangskontrollServiceImpl
import no.nav.modiapersonoversikt.legacy.sak.service.interfaces.TilgangskontrollService
import no.nav.modiapersonoversikt.service.pdl.PdlOppslagService
import no.nav.tjeneste.virksomhet.sakogbehandling.v1.binding.SakOgBehandlingV1
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class SakServiceConfig {
    @Bean
    open fun sakOgBehandlingService(porttype: SakOgBehandlingV1, pdl: PdlOppslagService): SakOgBehandlingService {
        return SakOgBehandlingServiceImpl(
            porttype,
            pdl
        )
    }

    @Bean
    open fun sakstemaService(): SakstemaService {
        return SakstemaService()
    }

    @Bean
    open fun tilgangskontrollService(): TilgangskontrollService? {
        return TilgangskontrollServiceImpl()
    }
}
